Sources/XCTHealthKit/XCTHealthKitAddSampleInput.swift

```diff
@@ -0,0 +1,224 @@
+public struct NewHealthSampleInput {
+    /// Helper type for inserting the sample's value
+    public struct EnterSampleValueHandler {
+        /// Signature of the actual value input function.
+        public typealias Imp = @MainActor (_ sample: NewHealthSampleInput, _ healthApp: XCUIApplication) throws -> Void
+        let imp: Imp
+    }
+    /// The quantity type for which the new sample should be added
+    public let sampleType: HealthAppSampleType
+    /// The date which should be used for the new sample. Specifying `nil` will use HealthKit's default date (i.e., the current date).
+    public let date: DateComponents?
+    /// The handler to which the actual entering of the sample value will be delegated.
+    public let enterSampleValueHandler: EnterSampleValueHandler
+    
+    /// - Warning: The `date` parameter is not guaranteed to actually always get properly applied when creating a sample.
+    public init(
+        sampleType: HealthAppSampleType,
+        date: DateComponents? = nil, // swiftlint:disable:this function_default_parameter_at_end
+        enterSampleValueHandler: EnterSampleValueHandler
+    ) {
+        self.sampleType = sampleType
+        self.date = date // nil // currently disabled, until we can figure out why the datePickers in the health app's add data sheet are so cursed.
+        self.enterSampleValueHandler = enterSampleValueHandler
+    }
+}
+
+
+extension NewHealthSampleInput.EnterSampleValueHandler {
+    /// Creates a new sample value input handler that inserts a numeric value into a text field.
+    /// - parameter textFieldPredicate: optional predicate allowing the caller to specify which exact text field the value should be inserted into
+    public static func enterSimpleNumericValue(_ value: Double, inTextField textFieldPredicate: NSPredicate? = nil) -> Self {
+        Self { _, healthApp in
+            let dataEntryTable = healthApp.tables["UIA.Health.AddData.View"]
+            let textField: XCUIElement
+            if let textFieldPredicate {
+                textField = dataEntryTable.textFields.matching(textFieldPredicate).firstMatch
+            } else {
+                textField = dataEntryTable.cells["UIA.Health.AddData.ValueCell"].textFields.firstMatch
+            }
+            XCTAssert(textField.waitForExistence(timeout: 2))
+            textField.tap()
+            let stringValue: String
+            if abs(value.rounded(.toNearestOrEven) - value) < 0.00001 {
+                // it's essentially an int
+                stringValue = String(Int(value.rounded(.toNearestOrEven)))
+            } else {
+                stringValue = String(format: "%.2f", locale: .current, value)
+            }
+            textField.typeText(stringValue)
+        }
+    }
+    
+    /// Creates a new sample value input handler that uses a custom, caller-provided closure to insert the sample's value into the Health app.
+    public static func custom(_ imp: @escaping Fn) -> Self {
```

Comment:
We will need to implement this path 👍

Reply:
not sure what you mean; i might be missing something but the `custom` function is fully implemented and in fact used (eg when inserting an ECG sample, iirc)

Reply:
Sorry, could have been clearer about this! I was pointing to the build error due to the missing `Fn` type.